custom/enikshay/integrations/ninetyninedots/utils.py

```diff
@@ -0,0 +1,167 @@
+import uuid
+import pytz
+from django.utils.dateparse import parse_datetime
+
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from casexml.apps.case.const import CASE_INDEX_EXTENSION, UNOWNED_EXTENSION_OWNER_ID
+from corehq.form_processor.exceptions import CaseNotFound
+from dimagi.utils.decorators.memoized import memoized
+
+from casexml.apps.case.mock import CaseFactory, CaseStructure, CaseIndex
+from custom.enikshay.integrations.ninetyninedots.exceptions import AdherenceException
+
+
+class AdherenceCaseFactory(object):
+    """
+    Creates and updates adherence cases for a given person
+
+    Only for use by the 99DOTS integration with eNikshay
+    """
+    DEFAULT_ADHERENCE_CONFIDENCE = "medium"
+    ADHERENCE_CASE_TYPE = "adherence"
+    DEFAULT_ADHERENCE_VALUE = "unobserved_dose"
+
+    def __init__(self, domain, person_id):
+        self.domain = domain
+        self.person_id = person_id
+
+        self.case_accessor = CaseAccessors(domain)
+        self.case_factory = CaseFactory(domain)
+
+    @property
+    @memoized
+    def _person_case(self):
+        try:
+            return self.case_accessor.get_case(self.person_id)
+        except CaseNotFound:
+            raise AdherenceException(message="No patient exists with this beneficiary ID")
+
+    @property
+    @memoized
+    def _default_adherence_confidence(self):
+        return self._episode_case.dynamic_case_properties().get(
+            'default_adherence_confidence', self.DEFAULT_ADHERENCE_CONFIDENCE
+        )
+
+    @property
+    @memoized
+    def _episode_case(self):
+        return get_open_episode_case(self.domain, self._person_case.case_id)
+
+    def create_adherence_cases(self, adherence_points, adherence_source):
+        return self.case_factory.create_or_update_cases([
```

Comment:
was there a reason to use `CaseFactory` over `CaseBlock`? (not saying it matters, more just curious)